code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -52,7 +52,12 @@
 					else
 						visible_message("<span class='danger'>[src] deflects the projectile!</span>", "<span class='userdanger'>You deflect the projectile!</span>")
 					playsound(src, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, 1)
-					return 0
+					if(!mind.martial_art.reroute_deflection)
+						return 0
+					else
+						P.firer = src
+						P.setAngle(rand(0, 360))//SHING
+						return BULLET_ACT_FORCE_PIERCE

```

Comment:
Why does this proc accept return values of both strings and integers >:(

Reply:
i swear i saw return 0 somewhere else in the code but i suppose i was mistaken, i can fix that in a little bit

Reply:
oh it isn't suppose to accept numbers, how funny.